packages/bootstrap-vue-next/src/components/BOffcanvas/BOffcanvas.vue

```diff
@@ -284,6 +276,15 @@ const hide = (trigger = '') => {
     emit('hide-prevented')
     return
   }
+
+  if (trigger === 'close') {
+    emit(trigger, event)
+  }
+  if (trigger === 'esc') {
+    emit(trigger, event)
+  }
+  emit('hide', event)
+
```

Comment:
```suggestion

if(event.defaultPrevented) {
    emit('hide-prevented')
    return
}
```

We need to ensure that you can prevent the closing when this event is triggered